engine/src/interpolation/plugin.cpp

```diff
@@ -0,0 +1,72 @@
+#include <glm/glm.hpp>
+
+#include <cubos/core/ecs/reflection.hpp>
+#include <cubos/core/reflection/external/glm.hpp>
+#include <cubos/core/reflection/type.hpp>
+
+#include <cubos/engine/fixed_step/plugin.hpp>
+#include <cubos/engine/interpolation/plugin.hpp>
+#include <cubos/engine/physics/plugin.hpp>
+#include <cubos/engine/physics/solver/plugin.hpp>
+#include <cubos/engine/transform/plugin.hpp>
+
+using namespace cubos::engine;
+
+CUBOS_REFLECT_IMPL(cubos::engine::Interpolated)
+{
+    return cubos::core::ecs::TypeBuilder<Interpolated>("cubos::engine::Interpolated").build();
+}
+
+void cubos::engine::interpolationPlugin(Cubos& cubos)
+{
+    cubos.depends(fixedStepPlugin);
+    cubos.depends(transformPlugin);
+    cubos.depends(physicsPlugin);
+    cubos.depends(physicsSolverPlugin);
+
+    cubos.component<Interpolated>();
+
+    cubos.observer("add Interpolated when Velocity is added")
+        .onAdd<Velocity>()
+        .without<Interpolated>()
+        .call([](Commands cmds, Query<Entity, const Velocity&, const Position&, const Rotation&, const Scale&> query) {
+            for (auto [ent, _, position, rotation, scale] : query)
+            {
+                cmds.add(ent, Interpolated{.previousPosition = position.vec,
+                                           .previousRotation = rotation.quat,
+                                           .previousScale = scale.factor,
+                                           .nextPosition = position.vec,
+                                           .nextRotation = rotation.quat,
+                                           .nextScale = scale.factor});
+            }
+        });
```

Comment:
I think this should be responsibility of the physics plugin. I.e., the physics plugin should depend on interpolation, and add this observer, nto the other way around.